skimage/morphology/_rolling_ball_cy.pyx

```diff
@@ -0,0 +1,220 @@
+import numpy as np
+cimport cython
+from libc.math cimport isnan, INFINITY
+from cython.parallel cimport prange
+
+from .._shared.fused_numerics cimport np_floats
+
+ctypedef np_floats DTYPE_FLOAT
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+cdef Py_ssize_t ind2ind(Py_ssize_t from_index, Py_ssize_t offset, Py_ssize_t[::1] from_shape, Py_ssize_t[::1] to_shape) nogil:
+    """ 
+    Convert the flat index of one array to a flat index of another array
+
+    The primary use case for this is if one array is a view into the other
+    array and the dimensions are unknown at compile time (and, hence, typed
+    memoryviews can't be used).
+
+
+    Parameters
+    ----------
+    from_index : intp
+        The index in the original array
+    offset : intp
+        The distance (in number of elements) between the origins 
+        of both arrays
+    from_shape : (N) ndarray
+        The shape of the original array
+    to_shape : (N) ndarray
+        The shape of the resulting array
+    
+    Returns
+    -------
+    to_index : intp
+        The index in the resulting array
+    """
+
+    cdef Py_ssize_t ndim = from_shape.shape[0]
+    cdef Py_ssize_t idx, modulo, pos
+
+    cdef Py_ssize_t multiple_kernel = 1
+    cdef Py_ssize_t multiple_img = 1
+    cdef Py_ssize_t to_index = offset
+
+    for idx in range(ndim-1, -1, -1):
+        modulo = from_index % (from_shape[idx] * multiple_kernel)
+        pos = modulo // multiple_kernel
+        multiple_kernel *= from_shape[idx]
+
+        to_index += pos * multiple_img
+        multiple_img *= to_shape[idx]
+
+    return to_index
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+def apply_kernel_nan(DTYPE_FLOAT[::1] img not None,
+                     DTYPE_FLOAT[::1] kernel not None,
+                     DTYPE_FLOAT[::1] ellipsoid_intensity not None,
+                     Py_ssize_t[::1] img_shape not None,
+                     Py_ssize_t[::1] padded_img_shape not None,
+                     Py_ssize_t[::1] kernel_shape not None,
+                     Py_ssize_t num_threads=0):
```

Comment:
You can read more about the magic number `0` here: https://github.com/scikit-image/scikit-image/pull/4851#discussion_r477223452

Essentially, it means to use OpenMP's `num_threads`, which typically is typically one thread per virtual core unless configured otherwise.